### PR TITLE
Refactor: Python formatting using Black and isort

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Python formatting using Black and isort

## Proposed changes

- Creating basic configuration for isort and Black on repo level
- Adding isort and black to precommit
- Running isort and black

**Review this PR one commit at a time to keep sanity :)**

## How to test

No behavoral changes. This should *only* be formatting.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -44,7 +50,7 @@ repos:
           - -d duplicate-code
 
   - repo: https://github.com/adrienverge/yamllint.git
-    rev: 'v1.23.0'
+    rev: "v1.23.0"
```
Comment (gmuloc) [inline .pre-commit-config.yaml:55]: Maybe let's take the opportunity in this PR to give nice names to all the hooks when running `pre-commit`:

```
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for merge conflicts................................................Passed
isort....................................................................Passed
black....................................................................Passed
Check for PEP8 error on Python files.....................................Passed
Check for Linting error on Python files..................................Passed
Check for Linting error on YAML files....................................Passed
Check for Linting error on Jinja2 templates..............................Passed
markdownlint.............................................................Passed
```

So for isort, black and markdownlint ?

File: ansible_collections/arista/avd/docs/contribution/style-guide.md [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+- [isort](https://pycqa.github.io/isort/) version 5.10.1
+- [Flake-8](https://flake8.pycqa.org/en/4.0.1/index.html) version 4.0.1
+- [pylint](https://pylint.pycqa.org/en/2.6/) version 2.6.0
+
+Configurations for above tools can be found in:
+
+- [pyproject.toml](../../../../../pyproject.toml)
+- [.pre-commit-config.yaml](../../../../../.pre-commit-config.yaml)
+- [.flake8](../../../../../.flake8)
+- [pylintrc](../../../../../pylintrc)
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/style-guide.md]: we could probably move flake8 and pylintrc in pyproject,toml as well I think?

File: ansible_collections/arista/avd/docs/contribution/style-guide.md [hunk 3]
```diff
@@ -1,12 +1,26 @@
 # AVD Coding Style
 
-This page provides a list of guidelines to apply when developing the Jinja2 template in the AVD context. These rules apply for either creation or update of any Jinja2 file available in a `aristanetworks/ansible-avd` repository.
+This page provides a list of guidelines to apply when developing Python code or Jinja2 template in the AVD context. These rules apply for either creation or update of any Python or Jinja2 file available in a `aristanetworks/ansible-avd` repository.
 
 ## Python code style
 
 As AVD is an Ansible collection, we're required to follow guidelines from the official [Ansible documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html) for all Python code.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/style-guide.md:7]: Let's state clearly here that we expect the line-length to be 160 to make it clear for people looking at this

File: ansible_collections/arista/avd/plugins/plugin_utils/strip_empties.py [hunk 4]
```diff
@@ -41,7 +42,15 @@ def strip_null_from_data(data, strip_values_tuple=(None,)):
     return data
 
 
-def strip_empties_from_list(data, strip_values_tuple=(None, "", [], {},)):
+def strip_empties_from_list(
+    data,
+    strip_values_tuple=(
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/strip_empties.py:47]: Maybe the default value can be a global var in this file given that it is used for two functions (though probably not as part of this PR)

File: ansible_collections/arista/avd/tests/unit/modules/test_configlet_build_config.py [hunk 5]
```diff
@@ -1,46 +1,47 @@
...
-from ansible_collections.arista.avd.plugins.modules.configlet_build_config import get_configlet
 import os
+
 import pytest
 
-CONFIGLETS_DIR = os.path.dirname(os.path.realpath(
-    __file__)) + "../../inventory/intended/configs"
+from ansible_collections.arista.avd.plugins.modules.configlet_build_config import get_configlet
+
+CONFIGLETS_DIR = os.path.dirname(os.path.realpath(__file__)) + "../../inventory/intended/configs"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/tests/unit/modules/test_configlet_build_config.py]:
```suggestion
CONFIGLETS_DIR = f"{os.path.dirname(os.path.realpath(__file__))}../../inventory/intended/configs"
```

File: ansible_collections/arista/avd/tests/unit/modules/test_inventory_to_container.py [hunk 6]
```diff
@@ -1,70 +1,62 @@
...
+            "DC1_SERVERS": {"devices": [], "parent_container": "DC1"},
+            "DC1_TENANTS_NETWORKS": {"devices": [], "parent_container": "DC1"},
+        },
+    },
+    "non_default_parent": {"parent": "CVP", "expected_output": {"CVP": {"parent_container": "Tenant"}}},
 }
 
-INVENTORY_FILE = os.path.dirname(
-    os.path.realpath(__file__)) + "/../../inventory/inventory.yml"
+INVENTORY_FILE = os.path.dirname(os.path.realpath(__file__)) + "/../../inventory/inventory.yml"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/tests/unit/modules/test_inventory_to_container.py]:
```suggestion
INVENTORY_FILE = f"{os.path.dirname(os.path.realpath(__file__))}/../../inventory/inventory.yml"
```

File: ansible_collections/arista/avd/docs/contribution/style-guide.md (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/contribution/style-guide.md]: I tried. Flake8 do not support pyproject.toml without a plugin.
Pylintrc is copied from Ansible, so I decided to leave it alone :)

File: ansible_collections/arista/avd/docs/contribution/style-guide.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/contribution/style-guide.md:7]: ack

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ClausHolbechArista) [reply .pre-commit-config.yaml:55]: done

File: ansible_collections/arista/avd/plugins/plugin_utils/strip_empties.py (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/strip_empties.py:47]: Not in this PR. It is busy enough as it is :)

**Comment by ClausHolbechArista** [comment]
The errors in mkdocs 404 checker are expected

File: ansible_collections/arista/avd/docs/contribution/style-guide.md (hunk 2 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/docs/contribution/style-guide.md]: Ack - resolving this

**Review by gmuloc — APPROVED** [review]
LGTM!

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**